openmc/model/model.py

```diff
@@ -238,6 +240,41 @@ def from_xml(cls, geometry='geometry.xml', materials='materials.xml',
         plots = openmc.Plots.from_xml(plots) if Path(plots).exists() else None
         return cls(geometry, materials, settings, tallies, plots)
 
+    @classmethod
+    def from_model_xml(cls, path='model.xml'):
+        """Create model from single XML file
+
+        Parameters
+        ----------
+        path : str or Pathlike
+            Path to model.xml file
+        """
+        tree = ET.parse(path)
+        root = tree.getroot()
+
+        model = cls()
+
+        model.settings = openmc.Settings.from_xml_element(root.find('settings'))
+        model.materials = openmc.Materials.from_xml_element(root.find('materials'))
+        materials = {str(m.id): m for m in model.materials}
+        model.geometry = openmc.Geometry.from_xml_element(root.find('geometry'), materials)
+
+        # gather meshses from other classes before reading the tally node
+        meshes = {}
+        if model.settings.entropy_mesh is not None:
+            meshes[model.settings.entropy_mesh.id] = model.settings.entropy_mesh
```

Comment:
There's also the UFS mesh (this feels a little brittle -- wondering if there's a better way we can capture all the meshes from settings.xml)
```suggestion
        if model.settings.entropy_mesh is not None:
            meshes[model.settings.entropy_mesh.id] = model.settings.entropy_mesh
        if model.settings.ufs_mesh is not None:
            meshes[model.settings.ufs_mesh.id] = model.settings.ufs_mesh
```

Reply:
Yeah maybe we should consolidate all of the mesh writing into a single method?